code/game/objects/items/weapons/shields.dm

```diff
@@ -183,6 +183,36 @@
 	icon_state = "som_shield"
 	soft_armor = list(MELEE = 35, BULLET = 50, LASER = 50, ENERGY = 50, BOMB = 30, BIO = 50, FIRE = 0, ACID = 15)
 
+//A shield that can be deployed as a barricade
+/obj/item/weapon/shield/riot/marine/deployable
+	name = "\improper TL-182 deployable shield"
+	desc = "A compact shield adept at blocking blunt or sharp objects from connecting with the shield wielder. Can be deployed as a barricade. Alt click to tighten the strap."
+	icon = 'icons/obj/items/weapons.dmi'
+	icon_state = "folding_shield"
+	flags_equip_slot = ITEM_SLOT_BACK
+	w_class = WEIGHT_CLASS_NORMAL
+	max_integrity = 300
+	integrity_failure = 50
+	soft_armor = list(MELEE = 35, BULLET = 30, LASER = 20, ENERGY = 40, BOMB = 25, BIO = 50, "rad" = 0, FIRE = 0, ACID = 30)
+	slowdown = 0.3
+	flags_item = IS_DEPLOYABLE
+	///The item this deploys into
+	var/deployable_item = /obj/structure/barricade/metal/deployable
+	///Time to deploy
+	var/deploy_time = 1 SECONDS
+	///Time to undeploy
+	var/undeploy_time = 1 SECONDS
+	///Whether it is wired
+	var/is_wired = FALSE
+
+/obj/item/weapon/shield/riot/marine/deployable/Initialize()
+	. = ..()
+	if(deployable_item)
```

Comment:
Why would deployable shit not have a deployable item

Reply:
Yeah they shouldn't unless someone fucks up I guess. I saw similar checks used for this element elsewhere though, but do you think I should just remove it?